wrapper_writer/scala_parser.py

```diff
@@ -0,0 +1,173 @@
+import re
+import yaml
+import os
+import glob
+import sys
+
+
+class ScalaParse:
```

Comment:
It might be worth pushing some of the generic functionality in here into a parent class as per the design